src/Backend/OnboardingInsuranceAPI/Extensions/StartupExtensions.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Extensions;
+
+internal static class StartupExtensions
+{
+    internal static IServiceCollection AddCosmosDb(this IServiceCollection services)
+    {
+        services.AddDbContext<DataContext>(optionsBuilder
+            => optionsBuilder.UseCosmos(
+            Environment.GetEnvironmentVariable("AccountEndpoint"),
+            Environment.GetEnvironmentVariable("AccountKey"),
+            databaseName: "Data",
+            options =>
+            {
+                options.ConnectionMode(ConnectionMode.Gateway);
+                options.WebProxy(new WebProxy());
+                options.Region(Regions.WestEurope);
+                options.GatewayModeMaxConnectionLimit(32);
+            })); //AddDbContextFactory try booth
+        return services;
+    }
+    internal static IServiceCollection AddInMemoryDb(this IServiceCollection services) //TODO implement mock data
+    {
+        services.AddDbContext<DataContext>(optionsBuilder
+            => optionsBuilder.UseInMemoryDatabase(
+                databaseName: "Data"
+            ));
+
+        //using (var context = new DataContext())
+        //{
+        //    context.Database.EnsureDeletedAsync();
+        //    context.Database.EnsureCreatedAsync();
+        //}
```

Comment:
This won't work like this, because the application is not yet finished building. You have to call it after you are done with building it and before you Run it.

Reply:
Ok